File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        s_l = set(len(i) for i in strs)
        ms = min(s_l)
        result = ''
        temp = strs[0]
        for i in range(0,ms):
            for j in strs:
                if temp[i] != j[i]:
                    return result
            
            result += temp[i]
        return result

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        
        StringBuilder st = new StringBuilder();
        int index = 0;
        String temp = strs[0];
        
        while (true) {
            for (String c : strs) {
                if (index >= c.length() || c.charAt(index) != temp.charAt(index)) {
                    return st.toString();
                }
            }
            st.append(temp.charAt(index));
            index++;
        }
    }
}